src/s1reader/s1_annotation.py

```diff
@@ -0,0 +1,316 @@
+        self.path_annotation=path_annotation #Accepted format: SAFE .zip file, SAFE directory, or annotation .xml file
+        self.polarization=polarization
+        self.subswath=subswath #Subswath of the annotation file to load
+        self._xml_loaded=None
+        self.kind=''
+
+        #Data access point for the annotation .xml file
+        self.xml_et=None
+
+    def _get_search_phrase(self):
+        str_safe=os.path.basename(self.path_annotation).replace('.zip','')
+        token_safe=str_safe.split('_')
+
+        phrase_base='{PLATFORM}-{MODE}{SUBSWATH}-{TYPE}-{POL}'.format(PLATFORM=token_safe[0].lower(),
+                MODE=token_safe[1].lower(),
+                SUBSWATH=self.subswath,
+                TYPE=token_safe[2].lower(),
+                POL=self.polarization.lower())
+
+        if self.kind=='calibration':
+            phrase_xml_to_search='calibration-{PLATFORM}-{MODE}{SUBSWATH}-{TYPE}-{POL}'.format(
+                PLATFORM=token_safe[0].lower(),
+                MODE=token_safe[1].lower(),
+                SUBSWATH=self.subswath,
+                TYPE=token_safe[2].lower(),
+                POL=self.polarization.lower())
+        elif self.kind=='noise':
+            phrase_xml_to_search='noise-{PLATFORM}-{MODE}{SUBSWATH}-{TYPE}-{POL}'.format(
+                PLATFORM=token_safe[0].lower(),
+                MODE=token_safe[1].lower(),
+                SUBSWATH=self.subswath,
+                TYPE=token_safe[2].lower(),
+                POL=self.polarization.lower())
+        elif self.kind=='':
+            phrase_xml_to_search=f'/{phrase_base}'
+        else:
+            raise ValueError(f'Cannot recognize the annotation type {self.kind}')
+
+        return phrase_xml_to_search
+
+
+    def _load_xml_as_et(self): #kind: ['', 'calibration', 'noise']
+        if os.path.isfile(self.path_annotation):
+            if self.path_annotation.endswith('.zip'):
+                zip_in=zipfile.ZipFile(self.path_annotation,'r')
+                phrase_xml_in_zip=self._get_search_phrase()
+
+                #search for the .xml file
+                filename_xml=None
+                for fileinfo in zip_in.filelist:
+                    if (phrase_xml_in_zip in fileinfo.filename) and (fileinfo.filename.endswith('.xml')):
+                        filename_xml=fileinfo.filename
+                        break
+                
+                #load the xml if found.
+                if filename_xml is None:
+                    raise ValueError('Cannot find the annotation file.')
+                else:
+                     self.xml_et=ET.fromstring(zip_in.read(filename_xml))
+                
+                #finalize
+                self._xml_loaded=filename_xml
+                zip_in.close()
+
+
+            elif self.path_annotation.endswith('.xml'):
+                #load the xml as file
+                self.xml_et=ET.parse(self.path_annotation)
+                self._xml_loaded=self.xml_et
```

Comment:
What happens if it doesn't end with `.zip` or `.xml`?

Reply:
I've added couple of lines to raise `ValueError` in that case. Nice catch.
https://github.com/seongsujeong/s1-reader/blob/7e85dd7cda8d2ff3d36779894b9e0c52365d6149/src/s1reader/s1_annotation.py#L83-L84
```python
else:
    raise ValueError(f'Invalid path for the annotation: {self.path_annotation}')
```